SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskProcessCallable.java

```diff
@@ -72,6 +81,13 @@ public String toString() {
 
   private boolean runHealthcheck(Process process) {
     Optional<HealthcheckOptions> maybeOptions = task.getTaskDefinition().getHealthcheckOptions();
+    if (maybeOptions.isPresent() && maybeOptions.get().getExpectHealthcheckConfigFile()) {
```

Comment:
Can we change this up a bit. The need for `expectHealthcheckConfigFile` doesn't solve the problem we're having. Right now we have the issue that we don't know at deploy time (when the SingularityDeploy is created) if the underlying things being deployed will or won't have file healthchecks implemented. So, I was thinking this would be more of a check like 'if healthcheck config file present -> run v2 -> else run v1'

Reply:
I was considering that but saw it as an opportunity for falsely passing health checks in the case that a config file doesn't appear in time. Is the timing of that something we're comfortable making a guarantee about?

My thinking is that this flag would tell the executor to fail health checks if the task doesn't provide a configuration when it should, and eliminate the false pass scenario.